styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesGCPServiceAccountSecretManagerTest.java

```diff
@@ -429,6 +429,23 @@ public void shouldHandleTooManyKeysCreated() throws IOException {
             SERVICE_ACCOUNT));
   }
 
+  @Test
+  public void shouldHandleTooManyKeysCreatedFromFailedPrecondition() throws IOException {
+    when(serviceAccountKeyManager.serviceAccountExists(anyString())).thenReturn(true);
+
+    final GoogleJsonResponseException resourceExhausted = new GoogleJsonResponseException(
+        new HttpResponseException.Builder(400, "Precondition check failed", new HttpHeaders()),
+        new GoogleJsonError().set("status", "FAILED_PRECONDITION"));
+
+    doThrow(resourceExhausted).when(serviceAccountKeyManager).createJsonKey(any());
+    doThrow(resourceExhausted).when(serviceAccountKeyManager).createP12Key(any());
```

Comment:
Interesting that mockito doesn't complain we are over stubbing here. I think if we fail to create json key, we will not go and create p12 key.

Reply:
:shrug: I just copy and modified the test from above